src/main/java/io/aiven/kafka/auth/json/JsonUtils.java

```diff
@@ -0,0 +1,87 @@
+import java.util.Collection;
+import java.util.List;
+import java.util.regex.Pattern;
+
+import com.google.gson.GsonBuilder;
+import com.google.gson.JsonDeserializationContext;
+import com.google.gson.JsonDeserializer;
+import com.google.gson.JsonElement;
+import com.google.gson.JsonIOException;
+import com.google.gson.JsonSyntaxException;
+import com.google.gson.reflect.TypeToken;
+
+public class JsonUtils {
+
+    private static final GsonBuilder GSON_FACTORY =
+        new GsonBuilder()
+            .registerTypeAdapter(Pattern.class, new RegexpJsonDeserializer());
+
+    public static List<AivenAcl> readAclConfiguration(final Path configFile)
+        throws JsonParseException {
+        try (final Reader reader = Files.newBufferedReader(configFile)) {
+            final Type t = new TypeToken<Collection<AivenAcl>>() {}.getType();
+            return GSON_FACTORY.create().fromJson(reader, t);
+        } catch (final JsonSyntaxException | JsonIOException | IOException ex) {
+            throw new JsonParseException(
+                String.format(
+                    "Failed to read acl configuration file: %s",
+                    configFile
+                ), ex);
+        }
+    }
+
+    public static List<AivenKafkaPrincipalMapping> readPrincipalMappings(final Path configFile)
+        throws JsonParseException {
+        try (final Reader reader = Files.newBufferedReader(configFile)) {
+            final Type t = new TypeToken<Collection<AivenKafkaPrincipalMapping>>() {}.getType();
+            return GSON_FACTORY.create().fromJson(reader, t);
+        } catch (final JsonSyntaxException | JsonIOException | IOException ex) {
+            throw new JsonParseException(
+                String.format(
+                    "Failed to read kafka principal mapping configuration file: %s",
+                    configFile
+                ), ex);
+        }
+    }
+
+    public static List<UserPassword> readUserPassword(final Path configFile)
+        throws JsonParseException {
+        try (final Reader reader = Files.newBufferedReader(configFile)) {
+            final Type t = new TypeToken<Collection<UserPassword>>() {}.getType();
+            return GSON_FACTORY.create().fromJson(reader, t);
+        } catch (final JsonSyntaxException | JsonIOException | IOException ex) {
+            throw new JsonParseException(
+                String.format(
+                    "Failed to read user/password configuration file: %s", configFile
+                ), ex);
+        }
+    }
+
+    public static class RegexpJsonDeserializer implements JsonDeserializer<Pattern> {
+        @Override
+        public Pattern deserialize(final JsonElement jsonElement,
+                                   final Type type,
+                                   final JsonDeserializationContext jsonDeserializationContext)
+            throws JsonParseException {
+            return !jsonElement.isJsonNull() ? Pattern.compile(jsonElement.getAsString()) : null;
```

Comment:
Is returning `null` the expected response for an invalid input (as opposes to throwing)? The documentation assumes throwing

Reply:
yes e.g. `AivenAcl`:
- resourceRe - could be null
- resourceRePattern - could be null ...

Reply:
Ah, right, `null` is valid here.